views/build/grunt/portableelement.js

```diff
@@ -0,0 +1,219 @@
+                excludeShallow: ['mathJax'],
+                exclude: ['qtiCustomInteractionContext', 'qtiInfoControlContext'],
+                paths: {
+                    'taoQtiItem': root + '/taoQtiItem/views/js',
+                    'qtiCustomInteractionContext': root + '/taoQtiItem/views/js/runtime/qtiCustomInteractionContext',
+                    'qtiInfoControlContext': root + '/taoQtiItem/views/js/runtime/qtiInfoControlContext'
+                }
+            }
+        }
+    });
+
+    /**
+     * Get the name of entry point for the portable element
+     * @param {Object} pciRuntimeData - the runtime object from the portable element manifest
+     * @param {String} prefix - the prefix identifier of the portable element
+     * @returns {String}
+     */
+    function getHookFileName(pciRuntimeData, prefix) {
+        if (Array.isArray(pciRuntimeData.src) && pciRuntimeData.src.length > 0) {
+            //by convention the first module is the hook file
+            return pciRuntimeData.src[0]
+                .replace(/\.js$/i, '')
+                .replace(/^\.\//, prefix + '/');
+        }
+    }
+
+    /**
+     * Get the name of the min file for the portable element
+     * @param {Object} pciRuntimeData - the runtime object from the portable element manifest
+     * @returns {String}
+     */
+    function getMinHookFile(pciRuntimeData) {
+        var minHookFile;
+        if (pciRuntimeData.hook) {
+            minHookFile = pciRuntimeData.hook;
+        } else if (Array.isArray(pciRuntimeData.libraries) && pciRuntimeData.libraries.length > 0) {
+            //by convention the first module is the min file
+            minHookFile = pciRuntimeData.libraries[0];
+        }
+        if(minHookFile){
+            return minHookFile.replace(/^\.\//, '');
+        }
+    }
+
+    /**
+     * Print report when all promises are resolved/rejected
+     * @param {Array|Function} report
+     */
+    function printReport(report) {
+        if(Array.isArray(report)){
+            report.forEach(function (r) {
+                printReport(r);
+            });
+        }else if(typeof report === 'function'){
+            report.call();
+        }
+    }
+
+    /**
+     * Get the portable element model from its manifest file
+     * @param file
+     * @returns {*}
+     */
+    function getPortableModelFromFile(file){
+        var model;
+        portableModels.forEach(function(portableModel){
+            if(file.match(new RegExp('\/' + portableModel.file + '$'))){
+                model = {};
+                model.type = portableModel.type;
+                model.manifest = grunt.file.readJSON(file);
+                model.basePath = file.replace('/' + portableModel.file, '');
+                model.id = model.manifest.typeIdentifier;
+                model.runtimeHook = getHookFileName(model.manifest.runtime, model.id);
+                model.minRuntimeFile = getMinHookFile(model.manifest.runtime);
```

Comment:
You might need to break the loop (ie `return false`)

Reply:
ok, done